src/OpenSage.Core/StringConverter.cs

```diff
@@ -0,0 +1,48 @@
+﻿namespace OpenSage.Core
+{
+    /// <summary>
+    /// Convert a printf formatted string to a C# formatted string
+    /// </summary>
+    public class StringConverter
+    {
+        //TODO: implement this properly
+        public static string FromPrintf(string printf)
+        {
+            string result = "";
```

Comment:
Use a `StringBuilder` here. Appending to a string causes a new allocation for every single character.